core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1822,6 +1822,11 @@ final class NameResolverErrorHandler implements Runnable {
         @Override
         public void run() {
           handleErrorInSyncContext(error);
+          if (defaultServiceConfig != null) {
+            realChannel.updateConfigSelector(defaultServiceConfig.getDefaultConfigSelector());
```

Comment:
This sets the default service config in all cases, not just if the initial name resolution fails. If we've already updated the service config, we want to continue using it. When we update the config selector, we should also be changing `lastServiceConfig`.

Reply:
Fixed and moved it to handleErrorInSyncContext()

Reply:
Yes we need lastServiceConfig = defaultServiceConfig; 
The value of lastServiceConfig is used [here](https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/ManagedChannelImpl.java#L1773).
To make the default service config apply only when initial name resolution fails, i.e. only when the service config from the name resolver had never be applied, we should change 
`if (defaultServiceConfig != null) `
to
`if (lastServiceConfig != EMPTY_SERVICE_CONFIG && defaultServiceConfig != null) `